torchrl/collectors/collectors.py

```diff
@@ -795,11 +795,14 @@ def _step_and_maybe_reset(self) -> None:
                 dtype=torch.bool,
             )
```

Comment:
```suggestion
            td_reset = td_reset.exclude("_reset")
```

Does it make sense to exlude the _reset key here in order not to copy it in  self._tensordict

Reply:
exclude is IMO far more expensive than cloning, as it's cpu-bound (clone is gpu-bound)
and for highly nested tds it will slow things down 
On my machine
![image](https://github.com/pytorch/rl/assets/25529882/73f432ce-34dd-42b5-b64e-fd7d393a8858)

Reply:
right, so what the idea wast is that _reset is actually being copied into self._tensordict for no reason so I was just looking for a way to avoid that